ray_provider/xcom/ray_backend.py

```diff
@@ -238,6 +281,56 @@ def on_success_callback(context):
 
                     RayBackend.get_hook().cleanup(handles=handles)
 
+    @staticmethod
+    def persist_kvstore(gcs_conn_id, gcs_bucket, keypath):
+        log.info("Persisting")
+        
+        def persist_gcs(gcs_conn_id, gcs_bucket, keypath, data):
+            from airflow.providers.google.cloud.hooks.gcs import GoogleCloudStorageHook
+            import json
+            import pickle 
+            try:
+                print("Serializing")
+                output = pickle.dumps(data)
+                gcs = GoogleCloudStorageHook(gcs_conn_id)
+                print("Serialized")
+
+                with NamedTemporaryFile('w') as tmp:
+                    print("Writing")
+                    tmp.write(output)
+                    print("Flushing")
+                    tmp.flush()
+                    print("Uploading")
+                    gcs.upload(
+                        bucket=gcs_bucket,
+                        object=keypath,
+                        filename=tmp.name
+                    )
+                    print("Persisted to %s" % tmp.name)
+                    return tmp.name
+            except Exception as e:
+                print("Error thrown peristing %s" % e)
+
+        ray_store = get_or_create_kv_store(
+            identifier=RayBackend.store_identifier
+        )
+        log.info("Got Store")
+
+        data = ray_store.get_store(ray_store.identifier)
+
+        if data:
+            log.info("Got data")
+            ret_str = ray_store.execute(
+                persist_gcs,
+                args=(gcs_conn_id, gcs_bucket, keypath, data),
+                kwargs={"ray_conn_id": "ray_cluster_connection"},
+                eager=False
+            )
```

Comment:
Hmm, how about let's just implement this as an actor method. I can push a fix here.